src/pyhf/infer/test_statistics.py

```diff
@@ -2,6 +2,48 @@
 from .mle import fixed_poi_fit, fit
 from ..exceptions import UnspecifiedPOI
 
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def _qmu_like(mu, data, pdf, init_pars, par_bounds):
+    """
+    Clipped version of _tmu where _qmu = 0 if muhat > 0 else _tmu
+
+    If the lower bound of the POI is 0 this automatically implments
+    qmu_tilde. Otherwise this is qmu (no tilde).
+    """
+    tensorlib, optimizer = get_backend()
+    tmu_stat, (mubhathat, muhatbhat) = _tmu_like(
+        mu, data, pdf, init_pars, par_bounds, return_fitted_pars=True
+    )
+    qmu = tensorlib.where(
+        muhatbhat[pdf.config.poi_index] > mu, tensorlib.astensor(0.0), tmu_stat
+    )
+    return qmu
+
+
+def _tmu_like(mu, data, pdf, init_pars, par_bounds, return_fitted_pars=False):
+    """
+    Basic Profile Likelihood statistic.
+
+    If the lower bound of the POI is 0 this automatically implments
+    tmu_tilde. Otherwise this is tmu (no tilde).
+    """
+    tensorlib, optimizer = get_backend()
+    mubhathat, fixed_poi_fit_lhood_val = fixed_poi_fit(
+        mu, data, pdf, init_pars, par_bounds, return_fitted_val=True
+    )
+    muhatbhat, unconstrained_fit_lhood_val = fit(
+        data, pdf, init_pars, par_bounds, return_fitted_val=True
+    )
+    tmu = fixed_poi_fit_lhood_val - unconstrained_fit_lhood_val
+    tmu = tensorlib.clip(tmu, 0, max_value=None)
+    if return_fitted_pars:
+        return tmu, (mubhathat, muhatbhat)
+    return tmu
```

Comment:
```suggestion
    tmu_like = fixed_poi_fit_lhood_val - unconstrained_fit_lhood_val
    tmu_like = tensorlib.clip(tmu, 0, max_value=None)
    if return_fitted_pars:
        return tmu_like, (mubhathat, muhatbhat)
    return tmu_like
```

Same suggestion as above for `qmu_like`